File: issn_record.ts
import { verifyIdentifier } from "./utility.ts";
import { normalizeISSN, validateISSN } from "./issn.ts";

/**
 * Verifies an ISSN using portal.issn.org database.
 * @param issn - The identifier string to verify.
 * @returns Promise<boolean> - True if the ISSN exists, otherwise false.
 */
export async function verifyISSN(issn: string): Promise<boolean> {
  const normalizedISSN = normalizeISSN(issn);
  return await verifyIdentifier(
    issn,
    `https://portal.issn.org/resource/ISSN/${
      encodeURIComponent(normalizedISSN)
    }`,
    validateISSN,
  );
}

File: issn_record.go
package metadatatools

import "net/url"

// VerifyISSN verifies an ISSN exists via portal.issn.org. The
// identifier is normalized before verifying.
func VerifyISSN(issn string) bool {
	normalized := NormalizeISSN(issn)
	return VerifyIdentifier(issn, "https://portal.issn.org/resource/ISSN/"+url.QueryEscape(normalized), ValidateISSN)
}
